base/mysqlxx/Pool.h

```diff
@@ -35,7 +36,9 @@ class Pool final
     struct Connection
     {
         mysqlxx::Connection conn;
-        int ref_count = 0;
+        /// Ref count modified in constructor/descructor of Entry
+        /// but also read in pool code.
+        std::atomic<int> ref_count = 0;
```

Comment:
Maybe also add relaxed memory order into all reads and writes?

Reply:
But, it's a real race if I understand right? We can make the wrong decision using relaxed memory order? But of course, thread sanitizer will shut up even with relaxed memory order. Also, it's just a connection pool for MySQL, there shouldn't be a lot of contention on this `ref_count`?

Reply:
> We can make the wrong decision using relaxed memory order?
Maybe... Ok, it is too dangerous to play with it.